hudi-common/src/main/java/org/apache/hudi/common/engine/HoodieReaderContext.java

```diff
@@ -479,6 +481,15 @@ public final UnaryOperator<T> projectRecord(Schema from, Schema to) {
     return projectRecord(from, to, Collections.emptyMap());
   }
 
+  /**
+   * Converts the comparable values in Comparables to the specific engine type
+   */
+  public final Comparable convertValueToEngineType(Comparable value) {
+    return value instanceof Comparables
```

Comment:
Maybe we just add a new method `convertOrderingValueToEngineType(OrderingValue)` so that we can get rid of the `instanceof` per record.

Reply:
This is taken care of in recent commits made by Danny